client/signup/steps/domains/index.jsx

```diff
@@ -61,6 +61,25 @@ const DomainsStep = React.createClass( {
 		return { products: productsList.get() };
 	},
 
+	componentWillMount: function() {
+		const { stepName, queryObject } = this.props;
+
+		// skip this step if domain name is given in the url via the `domain` query parameter
+		if ( queryObject && queryObject.domain ) {
+			SignupActions.saveSignupStep( {
+				stepName,
+				suggestion: {
+					domain_name: queryObject.domain,
+					product_slug: 'domain_reg'
```

Comment:
`domain_reg` is only the product slug for some domains (e.g. `.com`), so this will break for other TLDs. Also, we generally use `domainItem` ([here](https://github.com/Automattic/wp-calypso/blob/master/client/lib/cart-values/cart-items.js#L371)) for this to make it easy to change the underlying object in the future if we want.